cpu-miner.c

```diff
@@ -468,12 +468,25 @@ static bool gbt_work_decode(const json_t *val, struct work *work)
 		le32enc((uint32_t *)(cbtx+37), 0xffffffff); /* prev txout index */
 		cbtx_size = 43;
 		/* BIP 34: height in coinbase */
-		for (n = work->height; n; n >>= 8) {
-			cbtx[cbtx_size++] = n & 0xff;
-			if (n < 0x100 && n >= 0x80)
-				cbtx[cbtx_size++] = 0;
+		if (work->height < 17) {
+			/* Heights between 1 and 16 use OP_1 through OP_16. BIP34 uses
+			   "Serialized CScript" format, which usually has the first byte
+			   as the size of the integer to be pushed, followed by the
+			   integet itself. But "Serialized CSscript" format is different
```

Comment:
Honest question: does saying "serialized CScript" actually mandate the most compact encoding? Because you don't *have* to use OP0-OP16 to push those numbers onto the stack; any of the length-prefixed variants are just as valid in a script.

Reply:
You're right, BIP34 does not accurately describe the code. BIP34 should say the most compact "Serialized CScript" format, or something like that.